src/Workspaces/VisualBasic/Portable/FindSymbols/VisualBasicDeclaredSymbolInfoFactoryService.vb

```diff
@@ -127,171 +144,198 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.FindSymbols
                typeDecl.Members.Any() AndAlso
                typeDecl.Members.All(Function(m) TypeOf m Is TypeBlockSyntax) Then
 
-                declaredSymbolInfo = Nothing
-                Return False
+                Return
+            End If
+
+            If node.Kind() = SyntaxKind.PropertyBlock Then
+                node = DirectCast(node, PropertyBlockSyntax).PropertyStatement
+            ElseIf node.Kind() = SyntaxKind.EventBlock Then
+                node = DirectCast(node, EventBlockSyntax).EventStatement
+            ElseIf TypeOf node Is MethodBlockBaseSyntax Then
+                node = DirectCast(node, MethodBlockBaseSyntax).BlockStatement
             End If
 
             Dim kind = node.Kind()
             Select Case kind
                 Case SyntaxKind.ClassBlock, SyntaxKind.InterfaceBlock, SyntaxKind.ModuleBlock, SyntaxKind.StructureBlock
                     Dim typeBlock = CType(node, TypeBlockSyntax)
-                    declaredSymbolInfo = DeclaredSymbolInfo.Create(
+                    declaredSymbolInfos.Add(DeclaredSymbolInfo.Create(
                         stringTable,
                         typeDecl.BlockStatement.Identifier.ValueText,
                         GetTypeParameterSuffix(typeBlock.BlockStatement.TypeParameterList),
-                        GetContainerDisplayName(node.Parent),
-                        GetFullyQualifiedContainerName(node.Parent, rootNamespace),
+                        containerDisplayName,
+                        fullyQualifiedContainerName,
                         typeBlock.BlockStatement.Modifiers.Any(SyntaxKind.PartialKeyword),
                         If(kind = SyntaxKind.ClassBlock, DeclaredSymbolInfoKind.Class,
                         If(kind = SyntaxKind.InterfaceBlock, DeclaredSymbolInfoKind.Interface,
                         If(kind = SyntaxKind.ModuleBlock, DeclaredSymbolInfoKind.Module,
                         DeclaredSymbolInfoKind.Struct))),
-                        GetAccessibility(typeBlock, typeBlock.BlockStatement.Modifiers),
+                        GetAccessibility(container, typeBlock, typeBlock.BlockStatement.Modifiers),
```

Comment:
passing the container in greatly simplified accessibility logic in several of these blocks.